sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/channel/AnalyticsValidator.java

```diff
@@ -0,0 +1,131 @@
+package com.microsoft.appcenter.analytics.channel;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.VisibleForTesting;
+
+import com.microsoft.appcenter.analytics.ingestion.models.LogWithNameAndProperties;
+import com.microsoft.appcenter.analytics.ingestion.models.EventLog;
+import com.microsoft.appcenter.analytics.ingestion.models.PageLog;
+import com.microsoft.appcenter.channel.AbstractChannelListener;
+import com.microsoft.appcenter.ingestion.models.Log;
+import com.microsoft.appcenter.utils.AppCenterLog;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.microsoft.appcenter.analytics.Analytics.LOG_TAG;
+
+public class AnalyticsValidator extends AbstractChannelListener {
+
+    /**
+     * Max length of event/page name.
+     */
+    @VisibleForTesting
+    static final int MAX_NAME_LENGTH = 256;
+
+    /**
+     * Max length of properties.
+     */
+    @VisibleForTesting
+    static final int MAX_PROPERTY_ITEM_LENGTH = 125;
+
+    /**
+     * Max number of properties.
+     */
+    @VisibleForTesting
+    static final int MAX_PROPERTY_COUNT = 20;
+
+    @Override
+    public boolean shouldFilter(@NonNull Log log) {
+        if (log instanceof EventLog) {
```

Comment:
Works well, last comment though that I missed, since you introduced a base type, use it here to combine the 2 if conditions.

Reply:
This was done intentionally to exclude types like
```java
new LogWithNameAndProperties() {
    @Override
    public String getType() {
        return "";
    }
})
```